security/pkg/k8s/tokenreview/client.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokenreview
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	k8sauth "k8s.io/api/authentication/v1"
+)
+
+type specForJWTValidationRequest struct {
+	Token string `json:"token"`
+}
+
+type jwtValidationRequest struct {
+	APIVersion string                      `json:"apiVersion"`
+	Kind       string                      `json:"kind"`
+	Spec       specForJWTValidationRequest `json:"spec"`
+}
+
+// Client is the client to call the TokenReview API.
+type Client struct {
+	url         string
+	caCert      []byte
+	reviewerJWT string
+}
+
+// NewClient creates a new tokenReviewClient.
+func NewClient(url string, caCert []byte, reviewerJWT string) *Client {
+	return &Client{
+		url:         url,
+		caCert:      caCert,
+		reviewerJWT: reviewerJWT,
+	}
+}
+
+// Review calls the K8s TokenReview API to verify a JWT, and returns the
+// <namespace>:<serviceaccountname> in the JWT if successful.
+// targetJWT: the target JWT to be verified.
+func (c *Client) Review(targetJWT string) (string, error) {
```

Comment:
client.go is not needed: this PR should call the utility function ValidateK8sJwt() in #10672 to validate a k8s JWT.

Reply:
Actually I'm not in favor of putting the function into pkg/util. The better practice is to try to avoid using "util" directory because its purpose is vague.